File: csharp/src/api/arch/RoleGroup.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/14
*/
using System.Text.Json;
using Src.Base;

namespace Src.Api.Jdy;

public class RoleGroupApiClient : ApiClient
{
    private readonly static string[] validVersions = { "v2" };
    private readonly static string defaultVersion = "v2";
    public RoleGroupApiClient(string host, string apiKey, string version = "") : base(host, apiKey, version)
    {
    }

    public new async Task<JsonElement?> doRequest(IDictionary<string, object> option)
    {
        if (Array.IndexOf(validVersions, base.version) == -1)
        {
            base.version = defaultVersion;
        }
        return await base.doRequest(option);
    }


    /**
    * 角色组列表
    */
    public async Task<JsonElement?> roleGroupList(IDictionary<string, object> option)
    {
        IDictionary<string, object> requestOption = new Dictionary<string, object>();
        requestOption.Add("method", "POST");
        requestOption.Add("path", "role_group/list");
        requestOption.Add("payload", option);
        return await this.doRequest(requestOption);
    }

    /**
    * 创建角色组
    */
    public async Task<JsonElement?> roleGroupCreate(string name)
    {
        IDictionary<string, object> requestOption = new Dictionary<string, object>();
        requestOption.Add("method", "POST");
        requestOption.Add("path", "role_group/create");
        IDictionary<string, object> payload = new Dictionary<string, object>();
        payload.Add("name", name);
        requestOption.Add("payload", payload);
        return await this.doRequest(requestOption);
    }


    /**
    * 更新角色组
    */
    public async Task<JsonElement?> roleGroupUpdate(int roleGroupNo, string name)
    {
        IDictionary<string, object> requestOption = new Dictionary<string, object>();
        requestOption.Add("method", "POST");
        requestOption.Add("path", "role_group/update");
        IDictionary<string, object> payload = new Dictionary<string, object>();
        payload.Add("role_group_no", roleGroupNo);
        payload.Add("name", name);
        requestOption.Add("payload", payload);
        return await this.doRequest(requestOption);
    }

    /**
    * 删除自建角色组
    */
    public async Task<JsonElement?> roleGroupDelete(int roleGroupNo)
    {
        IDictionary<string, object> requestOption = new Dictionary<string, object>();
        requestOption.Add("method", "POST");
        requestOption.Add("path", "role_group/delete");
        IDictionary<string, object> payload = new Dictionary<string, object>();
        payload.Add("role_group_no", roleGroupNo);
        requestOption.Add("payload", payload);
        return await this.doRequest(requestOption);
    }
}

File: python/src/api/arch/role_group.py
from src.model.http.api_client import ApiClient
from ...constants.http_constant import HttpConstant
from ...model.http.http_request_param import HttpRequestParam

# 合法的版本
valid_versions = ('v5')
# 默认版本
default_version = 'v5'


class RoleGroupApiClient(ApiClient):

    def __init__(self, api_key, host):
        ApiClient.__init__(self, api_key, host, valid_versions, default_version)

    """
     列出角色组
     Arguments:
        role_list_query_param: RoleListQueryParam 实例
        version: 版本
    """

    def roleGroupList(self, role_group_list_query_param, version=default_version):
        path = HttpConstant.ROLE_GROUP_PATH.format(suffix='list', version=self.getValidVersion(version))
        request_param = HttpRequestParam(path, role_group_list_query_param)
        return self.send_post(request_param)

    """
    创建角色组
    Arguments:
        name: str 
        version: 版本
    """

    def roleGroupCreate(self, name, version=default_version):
        path = HttpConstant.ROLE_GROUP_PATH.format(suffix='create', version=self.getValidVersion(version))
        request_param = HttpRequestParam(path, {"name": name})
        return self.send_post(request_param)

    """
    修改角色组
    Arguments:
        name: str 
        role_group_no: int
        version: 版本
    """

    def roleGroupUpdate(self, name, role_group_no, version=default_version):
        path = HttpConstant.ROLE_GROUP_PATH.format(suffix='update', version=self.getValidVersion(version))
        request_param = HttpRequestParam(path, {"name": name, "role_group_no": role_group_no})
        return self.send_post(request_param)

    """
    删除角色组
    Arguments:
        role_group_no: int
        version: 版本
    """

    def roleGroupDelete(self, role_group_no, version=default_version):
        path = HttpConstant.ROLE_GROUP_PATH.format(suffix='delete', version=self.getValidVersion(version))
        request_param = HttpRequestParam(path, {"role_group_no": role_group_no})
        return self.send_post(request_param)
